src/GLX/glvnd_genentry.c

```diff
@@ -95,12 +95,25 @@ extern char glx_entrypoint_end[];
 
 #elif defined(USE_AARCH64_ASM)
 
+#if defined(__CHERI_PURE_CAPABILITY__)
+#define STUB_SIZE 32
+#define STUB_ASM_ARCH(slot) \
+    "adrp c16, entrypointFunctions + " slot "*16\n" \
+    "str c17, [csp, #-16]\n" \
+    "adrp c17, 0\n" \
+    "scvalue c17, c17, x16\n" \
```

Comment:
I don't understand this. c16 is the result of an adrp, so why do we need to do another adrp and scvalue it? The metadata will be identical.